docs/pkg/client/events.md

```diff
@@ -0,0 +1,376 @@
+cometWebsocketURL := "ws://example.com"
+evtClient := events.NewEventsQueryClient(cometWebsocketURL)
+
+// Subscribing to a specific event, e.g. newly committed blocks:
+// (see: https://docs.cosmos.network/v0.47/core/events#subscribing-to-events)
+observable := evtClient.EventsBytes(ctx, "tm.event='NewBlock'")
+
+// Subscribe and receive from the observer channel, typically in some other scope.
+observer := observable.Subscribe(ctx)
+
+// Observer channel closes when the context is canceled, observer is
+// unsubscribed, or after the subscription returns an error.
+for eitherEvent := range observer.Ch() {
+    // (see either.Either: https://github.com/pokt-network/poktroll/blob/main/pkg/either/either.go#L3)
+    eventBz, err := eitherEvent.ValueOrError()
+
+    // ...
+}
+```
+
+### Advanced Usage
+
+```go
+// Given some custom dialer & connection implementation, e.g.:
+var (
+    tcpDialer events.Dialer = exampletcp.NewTcpDialerImpl()
+    grcpDialer events.Dialer = examplegrpc.NewGrpcDialerImpl()
+)
+
+// Both TCP and gRPC use the TCP scheme as gRPC uses TCP for its transport layer.
+cometUrl = "tcp://example.com"
+
+// Creating new EventsQueryClients with a custom tcpDialer:
+tcpDialerOpt := events.WithDialer(tcpDialer)
+tcpEvtClient := events.NewEventsQueryClient(cometUrl, tcpDialerOpt)
+
+// Alternatively, with a custom gRPC dialer:
+gcpDialerOpt := events.WithDialer(grcpDialer)
+grpcEvtClient := events.NewEventsQueryClient(cometUrl, grpcDialerOpt)
+
+// ... rest follows the same as the basic example.
+```
+
+### Configuration
+
+- **WithDialer**: Configure the client to use a custom dialer for connections.
+
+## Usage (`EventsReplayClient`)
+
+### Basic Usage
+
+```go
+// Define a query string to provide to the EventsQueryClient
+// See: https://docs.cosmos.network/main/learn/advanced/events#subscribing-to-events
+// And: https://docs.cosmos.network/main/learn/advanced/events#default-events
+const eventQueryString = "tm.event='Tx' AND message.action='eventType'"
+
+// Define the websocket URL the EventsQueryClient will subscribe to
+const cometWebsocketURL = "ws://example.com:36657"
+
+// Define an interface to represent the onchain event
+type EventType interface {
+  GetName() string
+}
+
+// Define the event type that implements the interface
+type eventType struct {
+  Name string `json:"name"`
+}
+
+func (e *eventType) GetName() string { return e.Name }
+
+// Define a decoder function that can take the raw event bytes
+// received from the EventsQueryClient and convert them into
+// the desired type for the EventsReplayClient
+func newEventType(eventBz []byte) EventType {
+  eventMsg := new(eventType)
+  if err := json.Unmarshal(eventBz, eventMsg); err != nil {
+    return nil, err
+  }
+
+  // Confirm the event is correct by checking its fields
+  if eventMsg.Name == "" {
+    return nil, events.ErrEventsUnmarshalEvent.
+      Wrapf("unable to unmarshal eventType: %s", string(eventBz))
+  }
+
+  return eventMsg, nil
+}
```

Comment:
Wdyt about making this example use a factory function as that's a pattern that may prove useful in many consumer contexts? We can use a `context.Context` in this example but perhaps there are other and/or more idiomatic factory function parameters.

```suggestion
func eventTypeFactory(ctx context.Context) func ([]byte) EventType {
  return function(eventBz []byte) EventType {  eventMsg := new(eventType)
    logger := polylog.Ctx(ctx)
  
    if err := json.Unmarshal(eventBz, eventMsg); err != nil {
      return nil, err
    }

    // Confirm the event is correct by checking its fields
    if eventMsg.Name == "" {
      return nil, events.ErrEventsUnmarshalEvent.
        Wrapf("unable to unmarshal eventType: %s", string(eventBz))
    }

    return eventMsg, nil
  }
}
```

Reply:
Perhaps it would also be worth defining a `NewEventFn = func([]byte) EventType` generic (see suggestion above for usage).